swatch-product-configuration/src/main/java/com/redhat/swatch/configuration/registry/SubscriptionConfigRegistry.java

```diff
@@ -0,0 +1,242 @@
+ * lookup methods for that list.
+ */
+@Slf4j
+class SubscriptionConfigRegistry {
+
+  private static SubscriptionConfigRegistry instance = null;
+
+  @Getter private final List<Subscription> subscriptions;
+
+  public static synchronized SubscriptionConfigRegistry getInstance() {
+    if (instance == null) {
+      instance = new SubscriptionConfigRegistry();
+    }
+    return instance;
+  }
+
+  SubscriptionConfigRegistry() {
+    subscriptions = new ArrayList<>();
+    Constructor constructor = new Constructor(Subscription.class, new LoaderOptions());
+    constructor.getPropertyUtils().setSkipMissingProperties(true);
+
+    Yaml yaml = new Yaml(constructor);
+
+    // Get the class loader
+    ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+
+    // Get all resources on the classpath under the subscription_configs subdirectory.
+    String resourceDirectory = "subscription_configs";
+    URL resourceUrl = Objects.requireNonNull(classLoader.getResource(resourceDirectory));
+
+    Path directoryPath;
+
+    // Needed to support AppClassLoader for when a fancy Quarkus ClassLoader isn't available
+    if (Objects.equals(resourceUrl.getProtocol(), "jar")) {
+      FileSystem fileSystem;
+      try {
+        fileSystem = FileSystems.newFileSystem(resourceUrl.toURI(), Collections.emptyMap());
+      } catch (IOException | URISyntaxException e) {
+        throw new ConfigResourcesLoadingException(e);
+      }
+      directoryPath = fileSystem.getPath(resourceDirectory);
+    } else {
+      try {
+        directoryPath = Paths.get(resourceUrl.toURI());
+      } catch (URISyntaxException e) {
+        throw new ConfigResourcesLoadingException(e);
+      }
+    }
+
+    log.debug("Attempting to walk directoryPath: {}", directoryPath);
+
+    // List YAML files in the resource directory
+    try (Stream<Path> paths = Files.walk(directoryPath)) {
+      paths
+          .filter(Files::isRegularFile)
+          .filter(file -> file.getFileName().toString().endsWith(".yaml"))
+          .forEach(
+              file -> {
+                try (InputStream inputStream = Files.newInputStream(file)) {
+                  var subscriptionFromYaml = yaml.loadAs(inputStream, Subscription.class);
+                  subscriptions.add(subscriptionFromYaml);
+                } catch (IOException e) {
+                  throw new ConfigResourcesLoadingException(e);
+                }
+              });
+    } catch (IOException e) {
+      log.error("Error while listing files. {}", e.getMessage());
+      throw new ConfigResourcesLoadingException(e);
+    }
+  }
+
+  /**
+   * @param serviceType
+   * @return Optional<Subscription>
+   */
+  Optional<Subscription> lookupByServiceType(String serviceType) {
+
+    return subscriptions.stream()
+        .filter(subscription -> Objects.nonNull(subscription.getServiceType()))
+        .filter(subscription -> Objects.equals(subscription.getServiceType(), serviceType))
```

Comment:
I've removed the first filter() condition Objects.nonNull(subscription.getServiceType()) because Objects.equals() already handles null-checking. So, there's no need to explicitly filter out null values for the service type. 

```
subscriptions.stream()
    .filter(subscription -> Objects.equals(subscription.getServiceType(), serviceType))
    .findFirst();
```